src/resource_clients/dataset.ts

```diff
@@ -156,29 +123,75 @@ class DatasetClient extends ResourceClient {
         });
     }
 
-    /**
-     * @param response
-     * @return {PaginationList}
-     * @private
-     */
-    _createPaginationList(response) {
+    private _createPaginationList<D>(response: AxiosResponse<D[]>): PaginatedList<D> {
         return {
             items: response.data,
             total: Number(response.headers['x-apify-pagination-total']),
             offset: Number(response.headers['x-apify-pagination-offset']),
             count: response.data.length, // because x-apify-pagination-count returns invalid values when hidden/empty items are skipped
             limit: Number(response.headers['x-apify-pagination-limit']), // API returns 999999999999 when no limit is used
+            // TODO: will the API return this
+            desc: false,
         };
     }
 }
 
-module.exports = DatasetClient;
+type MaybeArray<T> = T | T[];
 
-/**
- * @typedef {object} PaginationList
- * @property {object[]} items - List of returned objects
- * @property {number} total - Total number of objects
- * @property {number} offset - Number of objects that were skipped
- * @property {number} count - Number of returned objects
- * @property {number} limit - Requested limit
- */
+export interface Dataset {
+    id: string;
+    name: string;
+    userId: string;
+    createdAt: string;
+    modifiedAt: string;
+    accessedAt: string;
+    itemCount: number;
+    cleanItemCount: number;
```

Comment:
Some old datasets (like 2 years ago) don't have this field, not sure if we want to just forget about it?

Reply:
Couldn't a migration be written for those? Otherwise, I can mark the field as possibly undefined but then the docs also need an update.

Reply:
i guess we should model it based on how it works now, 2 years is quite a long time, and if should not matter much anyway

Reply:
I agree to let it be always a number. I just run into it when testing some old saved dataset but only a tiny tiny fraction of users have these saved somewhere and actively using them.